c7n/resources/s3.py

```diff
@@ -746,8 +746,10 @@ def process_bucket(self, b):
                 encryption_statement["Sid"] = s["Sid"]
             except:
                 log.info("Bucket:%s doesn't have Sid" % b['Name'])
-
-            encryption_statement["Resource"] = s["Resource"]
+            try:
+                encryption_statement["Resource"] = s["Resource"]
```

Comment:
rather than a blanket try/escape a key check ala `if 'Resource' in s:` would do..